google/cloud/internal/curl_impl.cc

```diff
@@ -626,7 +627,11 @@ StatusOr<std::size_t> CurlImpl::ReadImpl(absl::Span<char> output) {
         static_cast<HttpStatusCode>(http_code_), {});
     TRACE_STATE() << ", status=" << status << ", http code=" << http_code_
                   << "\n";
-    if (!status.ok()) return status;
+
+    if (!internal::Contains(ignored_http_error_codes_, http_code_) &&
```

Comment:
I think it might be easier to revert all the `not`s at this point:

```cc
  if (status.ok()) return bytes_read;
  if (internal::Contains(ignored_http_error_codes_, http_code_)) return bytes_read;
  return status;
```